gammapy/maps/geom.py

```diff
@@ -570,90 +590,258 @@ def slice(self, idx):
-        """Create from vectors of longitude and latitude in degrees."""
-        return cls(tuple([lon, lat] + list(args)), **kwargs)
+    def _from_lonlat(cls, coords, coordsys=None, copy=False):
+        """Create a `~MapCoord` from a tuple of coordinate vectors.  The first
+        two elements of the tuple should be longitude and latitude in
+        degrees.
+
+        Parameters
+        ----------
+        coords : tuple
+            Tuple of `~numpy.ndarray`.
+
+        Returns
+        -------
+        coord : `~MapCoord`
+            A coordinates object.
+        """
+        if isinstance(coords, (list, tuple)):
+            coords_dict = OrderedDict([('lon', coords[0]),
+                                       ('lat', coords[1])])
+            for i, c in enumerate(coords[2:]):
+                coords_dict['axis{}'.format(i)] = c
+        else:
+            raise ValueError('Unrecognized input type.')
+
+        return cls(coords_dict, coordsys=coordsys, copy=copy,
+                   match_by_name=False)
 
     @classmethod
-    def from_skydir(cls, skydir, *args, **kwargs):
-        """Create from vector of `~astropy.coordinates.SkyCoord`."""
-        if skydir.frame.name in ['icrs', 'fk5']:
-            return cls.from_lonlat(skydir.ra.deg, skydir.dec.deg, *args,
-                                   coordsys='CEL', **kwargs)
-        elif skydir.frame.name in ['galactic']:
-            return cls.from_lonlat(skydir.l.deg, skydir.b.deg, *args,
-                                   coordsys='GAL', **kwargs)
+    def _from_skycoord(cls, coords, coordsys=None, copy=False):
+        """Create from vector of `~astropy.coordinates.SkyCoord`.
+
+        Parameters
+        ----------
+        coords : tuple
+            Coordinate tuple with first element of type
+            `~astropy.coordinates.SkyCoord`.        
+        coordsys : {'CEL', 'GAL', None}
+            Spatial coordinate system of output `~MapCoord` object.
+            If None the coordinate system will be set to the frame of
+            the `~astropy.coordinates.SkyCoord` object.        
+        """
+        skycoord = coords[0]
+        if skycoord.frame.name in ['icrs', 'fk5']:
+            coords = (skycoord.ra.deg, skycoord.dec.deg) + coords[1:]
+            coords = cls._from_lonlat(coords, coordsys='CEL', copy=copy)
+        elif skycoord.frame.name in ['galactic']:
+            coords = (skycoord.l.deg, skycoord.b.deg) + coords[1:]
+            coords = cls._from_lonlat(coords, coordsys='GAL', copy=copy)
         else:
             raise ValueError(
-                'Unrecognized coordinate frame: {}'.format(skydir.frame.name))
+                'Unrecognized coordinate frame: {}'.format(skycoord.frame.name))
+
+        if coordsys is None:
+            return coords
+        else:
+            return coords.to_coordsys(coordsys)
 
     @classmethod
-    def from_tuple(cls, coords, **kwargs):
+    def _from_tuple(cls, coords, coordsys=None, copy=False):
```

Comment:
copy is unused here. Should be passed along to the _from_lonlat calls, no?

Reply:
Yes `copy` should get passed through.  Fixing now.